code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -965,30 +1006,41 @@
 
 	return
 
-/mob/living/silicon/robot/proc/choose_icon(var/triesleft, var/list/module_sprites)
+/mob/living/silicon/robot/proc/choose_icon()
+	set category = "Robot Commands"
+	set name = "Choose Icon"
+
 	if(!module_sprites.len)
 		src << "Something is badly wrong with the sprite selection. Harass a coder."
 		return
+	if (icon_selected == 1)
+		verbs -= /mob/living/silicon/robot/proc/choose_icon
+		return
+
+	if (icon_selection_tries == -1)
+		icon_selection_tries = module_sprites.len+1
+
 
-	icon_selected = 0
-	src.icon_selection_tries = triesleft
 	if(module_sprites.len == 1 || !client)
 		if(!(icontype in module_sprites))
 			icontype = module_sprites[1]
+		if (!client)
+			return
 	else
-		icontype = input("Select an icon! [triesleft ? "You have [triesleft] more chance\s." : "This is your last try."]", "Robot", icontype, null) in module_sprites
+		icontype = input("Select an icon! [icon_selection_tries ? "You have [icon_selection_tries] more chance\s." : "This is your last try."]", "Robot", icontype, null) in module_sprites
```

Comment:
If icon_selection_tries is ever zero, we wouldn't even have an option to choose.

Reply:
doing